lib/generators/draft/resource/templates/controllers/controller.rb

```diff
@@ -73,86 +43,50 @@ def create
   end
 
   def update
-    the_id = params.fetch(:rt_<%= singular_table_name %>_id)
+    the_id = params.fetch("id_from_path")
     @<%= singular_table_name.underscore %> = <%= class_name.singularize %>.where(:id => the_id).at(0)
 
 <% attributes.each do |attribute| -%>
-    @<%= singular_table_name.underscore %>.<%= attribute.column_name %> = params.fetch(:<%= attribute.column_name %>, @<%= singular_table_name.underscore %>.<%= attribute.column_name %>)
+<% if attribute.field_type == :check_box -%>
+    @<%= singular_table_name.underscore %>.<%= attribute.column_name %> = params.fetch("<%= attribute.column_name %>_from_query", false)
+<% else -%>
+    @<%= singular_table_name.underscore %>.<%= attribute.column_name %> = params.fetch("<%= attribute.column_name %>_from_query")
+<% end -%>
 <% end -%>
 
 <% unless skip_validation_alerts? -%>
     if @<%= singular_table_name.underscore %>.valid?
       @<%= singular_table_name.underscore %>.save
-      respond_to do |format|
-        format.json do
-          render({ :json => @<%= singular_table_name.underscore %>.as_json })
-        end
-  
-        format.html do
-          redirect_to("/<%= @plural_table_name.underscore %>/#{@<%= singular_table_name.underscore %>.id}", {:notice => "<%= singular_table_name.humanize %> updated successfully."})
-        end
-      end
+      redirect_to("/<%= @plural_table_name.underscore %>/#{@<%= singular_table_name.underscore %>.id}", { :notice => "<%= singular_table_name.humanize %> updated successfully."} )
     else
-      # render({:template => "/<%= @plural_table_name.underscore %>/edit_form_with_errors.html.erb"})
-      respond_to do |format|
-        format.json do
-          render({ :json => @<%= singular_table_name.underscore %>.as_json })
-        end
-  
-        format.html do
-          render({ :template => "<%= plural_table_name.underscore %>/show.html.erb" })
-        end
-      end
+      render({ :template => "<%= plural_table_name.underscore %>/show.html.erb" })
```

Comment:
```suggestion
     redirect_to("/<%= @plural_table_name.underscore %>/#{@<%= singular_table_name.underscore %>.id}", { :notice => "<%= singular_table_name.humanize %> failed to create successfully." })
```

Reply:
@raghubetina Should this message be an `:alert`, since it's a failure message?

Reply:
@jelaniwoods Yes, good idea.